source/grid/manifold_lib.cc

```diff
@@ -301,68 +301,146 @@ get_new_point (const std::vector<Point<spacedim> > &vertices,
 // ============================================================
 
 template <int dim, int spacedim>
-CylindricalManifold<dim,spacedim>::CylindricalManifold(const unsigned int axis,
-                                                       const double tolerance) :
-  direction (Point<spacedim>::unit_vector(axis)),
-  point_on_axis (Point<spacedim>()),
-  tolerance(tolerance)
+CylindricalManifold<dim, spacedim>::CylindricalManifold(const unsigned int axis,
+                                                        const double tolerance) :
+  ChartManifold<dim,spacedim,3>(Tensor<1,3>({0,2.*numbers::PI,0})),
+              normal_direction(compute_normal(Point<3>::unit_vector(axis))),
+              direction (Point<3>::unit_vector(axis)),
+              point_on_axis (Point<3>()),
+              tolerance(tolerance)
```

Comment:
This would actually be a good used case for a delegating constructor -- just forward to the other constructor.

(Not important, I just notice because we had that discussion today.)